sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -1490,5 +1490,161 @@ def pipeline_with_input(boolean: bool = False):
             .default_value.bool_value, True)
 
 
+class TestYamlComments(unittest.TestCase):
```

Comment:
These tests look good! A couple more test cases come to mind:
- compiling a Python component `@dsl.component`
- compiling a container component `@dsl.container_component`
- compiling a pipeline or component with inputs that don't have default values
- compiling a pipeline with an explicit `name` argument
- compiling a pipeline or component with a `float`, `int`, `dict`, and `list`
- compiling a pipeline or component with an artifact input and output

I don't think these each necessarily must be their own test (though that's fine too -- it's more specific but also more verbose). 

Bundling these characteristics together and doing full comment assertions (comparing the actual full comment against an expected full comment) could make tackling these many cases a bit more manageable and similar to real-world components we'll be handling.